src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/InviteUsers/Validation/InviteOrganizationUserValidator.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Bit.Core.AdminConsole.Errors;
+using Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers.InviteUsers.Models;
+using Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers.InviteUsers.Validation.PasswordManager;
+using Bit.Core.AdminConsole.Shared.Validation;
+using Bit.Core.Models.Business;
+using Bit.Core.OrganizationFeatures.OrganizationSubscriptions.Interface;
+using Bit.Core.Repositories;
+namespace Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers.InviteUsers.Validation;
+
+public interface IInviteUsersValidator : IValidator<InviteUserOrganizationValidationRequest>;
+
+public class InviteUsersValidator(
+    IOrganizationRepository organizationRepository,
+    IPasswordManagerInviteUserValidator passwordManagerInviteUserValidator,
+    IUpdateSecretsManagerSubscriptionCommand secretsManagerSubscriptionCommand) : IInviteUsersValidator
+{
+    public async Task<ValidationResult<InviteUserOrganizationValidationRequest>> ValidateAsync(InviteUserOrganizationValidationRequest request)
+    {
+        var subscriptionUpdate = new PasswordManagerSubscriptionUpdate(request);
+
+        var passwordManagerValidationResult = await passwordManagerInviteUserValidator.ValidateAsync(subscriptionUpdate);
+
+        if (passwordManagerValidationResult is Invalid<PasswordManagerSubscriptionUpdate> invalidSubscriptionUpdate)
+        {
+            return invalidSubscriptionUpdate.Map(request);
+        }
+
+        if (request.InviteOrganization.UseSecretsManager && request.Invites.Any(x => x.AccessSecretsManager))
+        {
+            return await ValidateSecretsManagerSubscriptionUpdateAsync(request, subscriptionUpdate);
+        }
+
+        return new Valid<InviteUserOrganizationValidationRequest>(new InviteUserOrganizationValidationRequest(
+            request,
+            subscriptionUpdate,
+            null));
+    }
+
+    private async Task<ValidationResult<InviteUserOrganizationValidationRequest>> ValidateSecretsManagerSubscriptionUpdateAsync(
+        InviteUserOrganizationValidationRequest request,
+        PasswordManagerSubscriptionUpdate subscriptionUpdate)
+    {
+        try
+        {
+            var smSubscriptionUpdate = new SecretsManagerSubscriptionUpdate(
+                    organization: await organizationRepository.GetByIdAsync(request.InviteOrganization.OrganizationId),
+                    plan: request.InviteOrganization.Plan,
+                    autoscaling: true)
+                .AdjustSeats(request.Invites.Count(x => x.AccessSecretsManager));
```

Comment:
This doesn't take into account any spare seats that the org may have. e.g. The org may be have 5 SM Seats in their sub, but only 4 users have access to SM, leaving 1 seat available. The invited user would take that seat without autoscaling.

You need to get the number of seats required: see [existing code](https://github.com/bitwarden/server/blob/46d36b1ef8a15b7c27654fe46ffa6aaa8c4b8e86/src/Core/AdminConsole/Services/Implementations/OrganizationService.cs#L865-L872) for reference. You were doing this with `InviteUserOrganizationValidationRequest.OccupiedSmSeats` but it got dropped somewhere I think.